### PR TITLE
feat: Add windows support

### LINKED ISSUE #206: Add windows documentation
**What would you like to be documented**:
We will need to write user documentation to describe how windows is supported. We will need to define all the edge cases and describe the [`preKubeadmCommands` commands ](https://github.com/oracle/cluster-api-provider-oci/pull/205#discussion_r1085152817)installed on the image from image builder (see PR link below).

**Why is this needed**:
Windows support will be added once https://github.com/kubernetes-sigs/image-builder/pull/1051 and https://github.com/oracle/cluster-api-provider-oci/pull/205 land

### PR DESCRIPTION
**Pull request by joekr**
**What this PR does / why we need it**:
Windows image builds are now supported using [Image-Builder v0.1.14](https://github.com/kubernetes-sigs/image-builder/releases/tag/v0.1.14). This adds Windows documentation and templates.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #206

### CODE REVIEW

File: docs/src/gs/create-windows-workload-cluster.md [hunk 1]
```diff
@@ -0,0 +1,198 @@
...
+1. Edit the `calico-vxlan.yaml` and modify the follow variables to allow Calico running on the nodes use VXLAN
+   - `CALICO_IPV4POOL_IPIP` - set to `"Never"`
+   - `CALICO_IPV4POOL_VXLAN` - set to `"Always"`
+   - `CALICO_IPV6POOL_VXLAN` - set to `"Always"`
+1. `kubectl apply -f calico-vxlan.yaml`
+1. Wait for the IPAMConfig to be loaded
+1. `kubectl patch IPAMConfig default --type merge --patch='{"spec": {"strictAffinity": true}}'`
+1. Edit the `calico-windows-vxlan.yaml` and modify the follow variables to allow Calico running on the nodes to talk
+to the Kubernetes control plane
+   - `KUBERNETES_SERVICE_HOST` - the IP address from step 5
```
Comment (shyamradhakrishnan) [inline docs/src/gs/create-windows-workload-cluster.md:152]: Step 5 does not have IP address related step right?

Comment (joekr) [reply docs/src/gs/create-windows-workload-cluster.md]: great catch. Updated

File: docs/src/gs/create-windows-workload-cluster.md [hunk 2]
```diff
@@ -0,0 +1,198 @@
...
+   - `CALICO_IPV4POOL_VXLAN` - set to `"Always"`
+   - `CALICO_IPV6POOL_VXLAN` - set to `"Always"`
+1. `kubectl apply -f calico-vxlan.yaml`
+1. Wait for the IPAMConfig to be loaded
+1. `kubectl patch IPAMConfig default --type merge --patch='{"spec": {"strictAffinity": true}}'`
+1. Edit the `calico-windows-vxlan.yaml` and modify the follow variables to allow Calico running on the nodes to talk
+to the Kubernetes control plane
+   - `KUBERNETES_SERVICE_HOST` - the IP address from the management cluster step
+   - `KUBERNETES_SERVICE_PORT`- the port from step the management cluster step
+   - `K8S_SERVICE_CIDR` - The VCN CIDR
```
Comment (shyamradhakrishnan) [inline docs/src/gs/create-windows-workload-cluster.md:154]: this is not VCN CIDR, this is service cidr we provide in the template

File: docs/src/gs/create-windows-workload-cluster.md [hunk 3]
```diff
@@ -0,0 +1,198 @@
...
+   - `DNS_NAME_SERVERS` -the IP address from step the management cluster step
```
Comment (shyamradhakrishnan) [inline docs/src/gs/create-windows-workload-cluster.md:155]: this should be the kube dns/core dns service ip

File: docs/src/gs/create-windows-workload-cluster.md (hunk 2 above)
Comment (joekr) [reply docs/src/gs/create-windows-workload-cluster.md]: updated

File: docs/src/gs/create-windows-workload-cluster.md (hunk 3 above)
Comment (joekr) [reply docs/src/gs/create-windows-workload-cluster.md]: updated

File: docs/src/gs/create-windows-workload-cluster.md [hunk 4]
```diff
@@ -0,0 +1,198 @@
...
+##### The steps to follow:
+
+**On the management cluster**
+
+1. Run `kubectl get OCICluster <cluster-name> -o jsonpath='{.spec.controlPlaneEndpoint.host}'` 
+   to get the `KUBERNETES_SERVICE_HOST` and `DNS_NAME_SERVERS` info that will be used in later steps
+
+**On the workload cluster**
+
+1. Download the [v3.24.5 calico release][calico-release]
```
Comment (shyamradhakrishnan) [inline docs/src/gs/create-windows-workload-cluster.md:140]: probably give the command here -curl -L https://github.com/projectcalico/calico/releases/download/v3.24.5/release-v3.24.5.tgz > release-v3.24.5.tgz

File: docs/src/gs/create-windows-workload-cluster.md [hunk 5]
```diff
@@ -0,0 +1,198 @@
...
+1. Uncompress the downloaded file and locate the `calico-vxlan.yaml`, `calico-windows-vxlan.yaml` and `windows-kube-proxy.yaml`
+files in the `manifests` dir
+1. Edit the `calico-vxlan.yaml` and modify the follow variables to allow Calico running on the nodes use VXLAN
+   - `CALICO_IPV4POOL_IPIP` - set to `"Never"`
+   - `CALICO_IPV4POOL_VXLAN` - set to `"Always"`
+   - `CALICO_IPV6POOL_VXLAN` - set to `"Always"`
```
Comment (shyamradhakrishnan) [inline docs/src/gs/create-windows-workload-cluster.md:146]: we dont need this right as we dont support IPV6 or atleast not tested till now

File: docs/src/gs/create-windows-workload-cluster.md [hunk 6]
```diff
@@ -0,0 +1,198 @@
...
+1. Wait for the IPAMConfig to be loaded
+1. `kubectl patch IPAMConfig default --type merge --patch='{"spec": {"strictAffinity": true}}'`
+1. Edit the `calico-windows-vxlan.yaml` and modify the follow variables to allow Calico running on the nodes to talk
+to the Kubernetes control plane
+   - `KUBERNETES_SERVICE_HOST` - the IP address from the management cluster step
+   - `KUBERNETES_SERVICE_PORT`- the port from step the management cluster step
+   - `K8S_SERVICE_CIDR` - The service CIDR set in the cluster template
+   - `DNS_NAME_SERVERS` - the IP address from dns service
+   - Change the namespace from `calico-system` to `kube-system`
+   - add `VXLAN_ADAPTER` env to the container named `node` the value should be `"Ethernet 2"`
```
Comment (shyamradhakrishnan) [inline docs/src/gs/create-windows-workload-cluster.md:157]: do we need this? I remember not doing this and it worked for me before?

File: docs/src/gs/create-windows-workload-cluster.md [hunk 7]
```diff
@@ -0,0 +1,198 @@
...
+1. `kubectl apply -f calico-windows-vxlan.yaml` (it takes a bit for this to pass livenessprobe)
+1. Edit the `windows-kube-proxy.yaml` 
+    - update the env `K8S_VERSION` to the version of kubernetes you are deploying
```
Comment (shyamradhakrishnan) [inline docs/src/gs/create-windows-workload-cluster.md:160]: specified in multple places, maybe better to specify the exact place, i.e.e under env variable

File: docs/src/gs/create-windows-workload-cluster.md [hunk 8]
```diff
@@ -0,0 +1,198 @@
...
+   - `CALICO_IPV6POOL_VXLAN` - set to `"Always"`
+1. `kubectl apply -f calico-vxlan.yaml`
+1. Wait for the IPAMConfig to be loaded
+1. `kubectl patch IPAMConfig default --type merge --patch='{"spec": {"strictAffinity": true}}'`
+1. Edit the `calico-windows-vxlan.yaml` and modify the follow variables to allow Calico running on the nodes to talk
+to the Kubernetes control plane
+   - `KUBERNETES_SERVICE_HOST` - the IP address from the management cluster step
+   - `KUBERNETES_SERVICE_PORT`- the port from step the management cluster step
+   - `K8S_SERVICE_CIDR` - The service CIDR set in the cluster template
+   - `DNS_NAME_SERVERS` - the IP address from dns service
```
Comment (shyamradhakrishnan) [inline docs/src/gs/create-windows-workload-cluster.md:155]: better to show the command for example  ```kubectl  get svc kube-dns -n kube-system  -o jsonpath='{.spec.clusterIP}'```

File: docs/src/gs/create-windows-workload-cluster.md (hunk 7 above)
Comment (joekr) [reply docs/src/gs/create-windows-workload-cluster.md]: updated

File: docs/src/gs/create-windows-workload-cluster.md (hunk 8 above)
Comment (joekr) [reply docs/src/gs/create-windows-workload-cluster.md:164]: updated

**Review by shyamradhakrishnan — APPROVED** [review]

**Pull request merged**